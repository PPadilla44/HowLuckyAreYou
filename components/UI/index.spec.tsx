import { cleanup, render, RenderAPI } from "@testing-library/react-native";
import { TextAsIcon } from "./index"

describe("<TextAsIcon />", () => {

    afterEach(cleanup);

    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(<TextAsIcon style={{fontSize: 48, fontWeight: "bold"}} text={"$"} />)
    })

    it("renders correctly", () => {
        const text = wrapper.getByText("$");
        expect(text).toHaveStyle({ fontSize: 48, fontWeight: "bold"});
    });

})
