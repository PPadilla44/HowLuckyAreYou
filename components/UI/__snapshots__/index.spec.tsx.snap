// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsIcon /> Matches Snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Array [
      Object {
        "backgroundColor": "#252525",
        "borderColor": "white",
      },
      Object {
        "alignItems": "center",
        "bottom": "5%",
        "height": 50,
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "width": 50,
      },
    ]
  }
  testID="settingsBtn"
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "#252525",
        "borderBottomWidth": 1,
        "borderColor": "white",
        "borderLeftWidth": 1,
        "borderRadius": 15,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
        "opacity": 1,
        "shadowColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View
        testID="settingsIcon"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 40,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "material",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            Óê©
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`<TextAsIcon /> Matches Snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "white",
        "fontFamily": "Futura",
      },
      Object {
        "fontSize": 48,
        "fontWeight": "bold",
      },
    ]
  }
>
  $
</Text>
`;
